docs/UserGuide.md

```diff
@@ -170,6 +173,955 @@ may have typed a command incorrectly as **RC4HDB** will warn you in the event th
+
+Format: `remove [/SPECIFIER] KEY/VALUE [ADDITIONAL_KEYS/ADDITIONAL_VALUES]`
+
+* A specifier is required in order for the command to work. If not it is an invalid command format
+* Currently, only two specifiers are supported:
+  * `/all` returns a resident if it fulfills **all** of the specified keywords.
+  * `/any` returns a resident if it fulfills **any** of the specified keywords.
+* Repeated keys are not permitted for both specifiers, e.g. `remove /all h/D h/A` will not work.
+* However, tags can be repeated in the command e.g. `remove /all t/exchange t/fresher`
+* Valid keys are those included [here](#format-for-resident-fields), and any additional tags.
+
+Examples:
+* `remove /all h/D g/M` deletes residents who are in Draco house, **and** are Male.
+* `remove /any h/D h/A` deletes residents belonging to either `Draco` **or** `Aquila` house.
+* `remove g/M` deletes residents who are male.
+
+[↑ Back to Top](#welcome-to-rc4hdb-user-guide)
+
+### Clearing all entries : `clear`
+
+---
+
+Clears all entries from **RC4HDB**.
+
+Format: `clear`
+
+<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
+Deleted data can not be retrieved. Do use this command cautiously!
+</div>
+
+[↑ Back to Top](#welcome-to-rc4hdb-user-guide)
+
+---
+
+### Format for resident fields
+
+`n/NAME`
+* Whitespaces are allowed *i.e. `n/Michael B. Jordan` is allowed*
+
+`p/PHONE_NUMBER`
+* Must be an **8**-digit non-negative integer
+
+`e/EMAIL`
+* Must follow the formatting for all standard emails *i.e. `e/Example@email.com` is accepted*
+* Can be both valid or invalid emails
+
+`r/FLOOR-UNIT`
+* The floor number and unit number must be separated by a hyphen
+* Both floor and unit number must be a non-negative integer *i.e. `5-8` is valid and `-3-8` is invalid*
+
+`g/GENDER`
+* `M` or `F`
+
+`h/HOUSE`
+* Represents the RC4 house that the resident is allocated to
+* Must be either `D`, `U`, `L`, `A`, `N`
+* `D` stands for **Draco**, `U` for **Ursa**, `L` for **Leo**, `A` for **Aquila**, `N` for **Noctua**
+
+`m/MATRIC_NUMBER`
+* Must be an uppercase `A`, followed by a **7**-digit non-negative integer and an uppercase alphabet. *i.e. `A0123456A`*
+
+`t/TAG`
+* Represents any other key that could be used to identify a resident
+* Must be a string. No restrictions on formatting
+* Optional. A resident can have any number of tags, including 0
+* When editing tags, the existing tags of the resident will be removed i.e adding of tags is not cumulative.
+* You can remove all the resident’s tags by typing `t/` without specifying any tags after it.
+
+[↑ Back to Top](#welcome-to-rc4hdb-user-guide)
+
+---
+
+## **Viewing Residents**
+
+You might find yourself overloaded with information. These commands can help you hide resident fields from view, 
+find specific residents, or search for residents whose fields match a specific keyword. By de-cluttering your screen, 
+these commands will allow you to focus only on the information you need!
```

Comment:
Very nice. But we could add one or more sentences to detail how this can help the user save time, or make their screens not as cluttered.